### PR TITLE
Support GitHub attestation endpoint

### PR DESCRIPTION
**Pull request by jhrozek**
In addition to fetching provenance attestations from an OCI image, let's also
support the GitHub attestation endpoint as a fallback (a fallback since it's
not GA yet).

The attestation reply is unmarshalled into a protobuf representation and
passed to sigstore-go.

Fixes: https://github.com/stacklok/epics/issues/174

### CODE REVIEW

**Comment by evankanderson** [comment]
I think your `Fixes` is not right...

File: internal/verifier/sigstore/container/container.go [hunk 1]
```diff
@@ -26,37 +26,96 @@ import (
...
+) ([]byte, []byte, error) {
+
 	// create a default verification result
 	params := newVerifyResult(BuildImageRef(registry, owner, artifact, version))
 
+	auth := newContainerAuth(authOpts...)
+
 	// construct the bundle
-	err := bundleFromOCIImage(params, newGithubAuthenticator(owner, accessToken))
+	// TODO: handle no access token?
```
Comment (JAORMX) [inline internal/verifier/sigstore/container/container.go:104]: I think we should error out, we should always have a token

**Comment by jhrozek** [comment]
> I think your `Fixes` is not right...

So we started filing the issues in the epics repo because it's private and we were not sure if we can reference the work publicly as the GH attestations program is not public. We got a green light from GH yesterday to merge the code in a public repo, but for some reason I can't seem to be able to move the issue to the minder repo. For now, I'm going to reference the full URL of [the issue](https://github.com/stacklok/epics/issues/174) in the epics repo. Is there a way to move the issue or do you know what can be preventing the issue from being moved? Is it that the epics repo is private but the minder repo is public?

File: internal/verifier/sigstore/container/container.go (hunk 1 above)
Comment (jhrozek) [reply internal/verifier/sigstore/container/container.go]: done, tested by running `mindev container verify` with no token.

**Review by rdimitrov — APPROVED** [review]
🚀

**Review by JAORMX — APPROVED** [review]
shipit :ship:

**Pull request merged**